pkg/build/builder/dockerutil.go

```diff
@@ -62,15 +73,35 @@ func pushImage(client DockerClient, name string, authConfig docker.AuthConfigura
 	if glog.V(5) {
 		opts.OutputStream = os.Stderr
 	}
-	var err error
+
 	var retriableError = false
 
+	eventWatch, err := eventClient.Watch(
```

Comment:
The image stream may actually be created during the push. Either when the push finishes successfully to hold the resulting image or when the push fails due to quota abuse - in order to reference something in the generated event.

There's no such event we're looking for before a PushImage call is invoked few lines below. We could achieve the same with just listing of the existing events afterwards but I'm not sure whether we can rely on the event being listed. Besides, isn't this more efficient compared to listing all of the events with the same filter? I'm assuming the watch returns just additions since the listing started.